lightningd/plugin_control.c

```diff
@@ -0,0 +1,135 @@
+#include <ccan/tal/path/path.h>
+#include <common/json_command.h>
+#include <common/jsonrpc_errors.h>
+#include <common/param.h>
+#include <dirent.h>
+#include <lightningd/plugin_control.h>
+#include <lightningd/plugin_hook.h>
+#include <unistd.h>
+
+/**
+ * A plugin command which permits to control plugins without restarting
+ * lightningd. It takes a command as first parameter (named 'control'
+ * to avoid ambuiguity), and eventually this control's parameter(s) as
+ * optional second and third parameters.
+ */
+static struct command_result *json_plugin_control(struct command *cmd,
+						const char *buffer,
+						const jsmntok_t *obj UNNEEDED,
+						const jsmntok_t *params)
+{
+	const char *control_command, *control_param, *control_second_param;
+	struct plugin *p;
+	struct json_stream *response;
+	bool plugin_found;
+
+	if (!param(cmd, buffer, params,
+		   p_req("command", param_string, &control_command),
+		   p_opt("parameter", param_string, &control_param),
+		   p_opt("second_parameter", param_string, &control_second_param),
+		   NULL))
```

Comment:
We probably need more powerful param handling for cases like this, TBH.  This will do for now, however.